### PR TITLE
#Task6 Andrey Yakshuk

### PR DESCRIPTION
**Pull request by andreyPromaster**
https://travis-ci.com/github/andreyPromaster/js-assignments/builds/175724254

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -166,7 +198,18 @@ function isInsideCircle(circle, point) {
  *   'entente' => null
  */
 function findFirstSingleChar(str) {
-    throw new Error('Not implemented');
+    for(let i=0; i<str.length; i++){
+        let coincidence = 0;
+        for(let j=0; j<str.length; j++){
+            if(str[i]==str[j]){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator `===` is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -166,7 +198,18 @@ function isInsideCircle(circle, point) {
...
+                coincidence++;
+            }
+        }
+        if(coincidence == 1){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator `===` is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -251,10 +312,27 @@ function reverseInteger(num) {
  *   4916123456789012 => false
  */
 function isCreditCardNumber(ccn) {
-    throw new Error('Not implemented');
+  let sum = 0;
+  let digits = String(ccn);
+  for (let i = 0; i < digits.length; i++) {
+    let cardNum = parseInt(digits[i]);
+
+    if ((digits.length - i) % 2 == 0) {
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator `===` is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -251,10 +312,27 @@ function reverseInteger(num) {
...
+
+      if (cardNum > 9) {
+        cardNum = cardNum - 9;
+      }
+    }
+
+    sum += cardNum;
+  }
+
+  return sum % 10 == 0;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator `===` is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -296,7 +385,20 @@ function getDigitalRoot(num) {
...
-    throw new Error('Not implemented');
+    let stack = [];
+    let opening_brackets = ['[','{','(','<'];
+    let closing_brackets = [']','}',')','>'];
+    for(let i = 0; i < str.length; i++){
+        if(opening_brackets.indexOf(str[i])>=0){
+            stack.push(opening_brackets.indexOf(str[i]));
+        }
+        else{   
+            if(closing_brackets[stack.pop()] != str[i]){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
"0" != 0
and "0" !== 0)

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -296,7 +385,20 @@ function getDigitalRoot(num) {
...
+                return false;
+            }
+        }
+    }
+    return stack.length == 0;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator `===` is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -332,7 +434,38 @@ function isBracketsBalanced(str) {
...
+        return `${Math.round(date/day)} days ago`;
+    else if(25*day <= date && date < 45*day)
+        return 'a month ago';
+    else if(45*day <= date && date < 345*day)
+        return `${Math.round(date/month)} months ago`;
+    else if(345*day <= date && date < 545*day)
+        return 'a year ago';
+    else if(546*day<=date)
+        return `${Math.round(date/year)} years ago`;
+
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, use structure if () {

}
every time.
It is usefull for readability

**Comment by andreyPromaster** [comment]
https://travis-ci.com/github/andreyPromaster/js-assignments/builds/178890089

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**